code/game/turfs/open/floor/catwalk_plating.dm

```diff
@@ -7,37 +7,30 @@
  */
 /turf/open/floor/catwalk_floor	//the base type, meant to look like a maintenance panel
 	icon = 'icons/turf/floors/catwalk_plating.dmi'
-	icon_state = "maint_below"
+	icon_state = "maint_above"
 	name = "catwalk floor"
 	desc = "Flooring that shows its contents underneath. Engineers love it!"
 	baseturfs = /turf/open/floor/plating
 	floor_tile = /obj/item/stack/tile/catwalk_tile
+	layer = CATWALK_LAYER
+	plane = GAME_PLANE
 	footstep = FOOTSTEP_CATWALK
 	overfloor_placed = TRUE
 	underfloor_accessibility = UNDERFLOOR_VISIBLE
 	var/covered = TRUE
-	var/above_state = "maint_above"	//Icon-state for the overlay
-
+	var/catwalk_type = "maint"
 
 /turf/open/floor/catwalk_floor/Initialize(mapload)
 	. = ..()
+	var/mutable_appearance/plating_underlay = mutable_appearance(icon, "[catwalk_type]_below", TURF_LAYER)
+	underlays += plating_underlay
```

Comment:
this is a bit sus, I'd think the same caching issue applies here

Reply:
Why underlays btb?

Reply:
I ditched the fancy caching since performance _shouldn't_ be critical for this code.  

For something like atmos code it makes sense to optimize it where we can.  But for catwalk initialization you might squeeze a fraction of a second during bootup times with the drawback making the code really ugly.

```
GLOBAL_LIST_EMPTY(catwalk_overlay_masterlist) //Stores all the above_states for the different types of catwalk

if(!GLOB.catwalk_overlay_masterlist[above_state])
		GLOB.catwalk_overlay_masterlist[above_state] = catwalk_overlay
	. += GLOB.catwalk_overlay_masterlist[above_state]
```

This is all really ugly and unnecessary just to give us a meager boost in performance where it doesn't matter. 

Also I suspect the reason caching was used before was that the overlay was being constantly created and deleted for each catwalk individually.  The masterlist created one overlay to be used for all catwalks.  

With my code refactor, catwalks have an underlay that is never deleted.  Only the icon_state is changed.  Technically, I could cache the underlays during init, but as I said before I think trading code cleanliness for performance is bad design.  (exceptions are performance bottlenecks)

Reply:
we cache overlays like that not just to save on cpu cycles, but also to save on memory (Though to be straight up with you I do not remember if overlays benefit from that in the same way as vis_contents)

Reply:
I posted in the comment sections the memory footprint for before and after.  The result was negligible.

Reply:
It's about the principle of the thing, unless there's a functional reason not to

Reply:
> When I was doing my testing, I noticed that the layers and planes would break only for overlays/underlays/mutable_appearances. Thus we only want to apply those updates to the main object directly. The way the code was before, the catwalk overlay was updating the plane/layer causing it to break. I reversed it so that the underlay is the bottom portion of the catwalk and the regular object is the upper portion with the grilles that switch on/off.


Quoted from above,

When I removed the `!GLOB.catwalk_overlay_masterlist` I did try to keep the plating as the main icon and update the overlay.  It wouldn't work.  So had to reverse their roles and make the main icon the catwalk grille and the underlay the plating.  Updating the layers/planes for overlays breaks the Multi Z when I tested this about a month ago.  

`Icons are for image manipulation`

All the manipulation is being performed on the icon, which is the catwalk plating top portion.  The bottom plating part never changes and is an underlay.

Reply:
Generating a new appearance for each catwalk when we only really ever need one copy seems silly, and I'm concerned about setting a good example on how to work with the things. I'm frankly not too sure if it has an impact, but I know if the pattern were continued elsewhere we'd have a headache, so I'd like to avoid it

Reply:
Let me know if https://github.com/timothymtorres/tgstation/commit/c16ba26e27fd0ac6d57f52afc0bb3c82551046b7 is satisfactory.